src/Controller/ResetPasswordControllerTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SymfonyCasts\Bundle\ResetPassword\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Provides useful methods to a "reset password controller"
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+trait ResetPasswordControllerTrait
+{
+    private function setCanCheckEmailInSession(Request $request, bool $value = true): void
+    {
+        $request->getSession()->set('ResetPasswordCheckEmail', $value);
+    }
+
+    private function isAbleToCheckEmail(SessionInterface $session): bool
```

Comment:
Same here - by using the request_stack service, we can avoid anything needing to be passed as an argument.

Reply:
done